packages/roosterjs-editor-api/lib/utils/blockFormat.ts

```diff
@@ -18,7 +23,17 @@ export default function blockFormat(
     editor.focus();
     editor.addUndoSnapshot((start, end) => {
         if (!beforeRunCallback || beforeRunCallback()) {
-            const regions = editor.getSelectedRegions();
+            const regions = editor.getSelectedRegions().filter(region => {
```

Comment:
when there is table selection, it means we can already treat each select table cell as a region. So it should be come simpler here.